pkgs/top-level/agda-packages.nix

```diff
@@ -8,10 +8,14 @@ let
       inherit Agda self;
       inherit (pkgs.haskellPackages) ghcWithPackages;
     }) withPackages mkDerivation;
-  in {
+  in rec {
     inherit mkDerivation;
 
-    agda = withPackages [] // { inherit withPackages; };
+    agdaLib = import ../build-support/agda/lib.nix {
+      inherit lib;
+    };
+
+    agda = withPackages [] // { inherit withPackages; lib = agdaLib; };
```

Comment:
```suggestion
    agda = withPackages [] // { inherit withPackages; };
```

Reply:
How do we access the function in `nixos/tests/agda.nix`?

Reply:
`pkgs.agdaPackages.agdaLib.interfaceFile`

Reply:
The whole purpose of adding `agda.lib` was to make `interfaceFile` accessible in tests, and maybe future derivations.

Reply:
Ok, forget my last comment, then.